nexus-watcher/src/events/processor.rs

```diff
@@ -117,7 +130,7 @@ impl EventProcessor {
     /// using the current cursor and a specified limit. It retrieves new event
     /// URIs in a newline-separated format, processes it into a vector of strings,
     /// and returns the result.
-    async fn poll_events(&mut self) -> Result<Option<Vec<String>>, DynError> {
+    async fn poll_events(&self) -> Result<Option<Vec<String>>, DynError> {
```

Comment:
Would it simplify things if instead of returning `Option<Vec<String>>` it would return `Vec<String>`?